webapp/components/FilterMenu/CategoriesFilter.spec.js

```diff
@@ -0,0 +1,80 @@
+import { mount } from '@vue/test-utils'
+import Vuex from 'vuex'
+import CategoriesFilter from './CategoriesFilter'
+
+const localVue = global.localVue
+
+let wrapper, environmentAndNatureButton, democracyAndPoliticsButton
+
+describe('CategoriesFilter.vue', () => {
+  const mutations = {
+    'posts/TOGGLE_CATEGORY': jest.fn(),
+    'posts/RESET_CATEGORIES': jest.fn(),
+  }
+  const getters = {
+    'posts/filteredCategoryIds': jest.fn(() => []),
+  }
+
+  const mocks = {
+    $t: jest.fn((string) => string),
+  }
+
+  const Wrapper = () => {
+    const store = new Vuex.Store({ mutations, getters })
+    const wrapper = mount(CategoriesFilter, { mocks, localVue, store })
+    wrapper.setData({
+      categories: [
+        { id: 'cat4', name: 'Environment & Nature', icon: 'tree', slug: 'environment-nature' },
+        {
+          id: 'cat15',
+          name: 'Consumption & Sustainability',
+          icon: 'shopping-cart',
+          slug: 'consumption-sustainability',
+        },
+        {
+          id: 'cat9',
+          name: 'Democracy & Politics',
+          icon: 'university',
+          slug: 'democracy-politics',
+        },
+      ],
+    })
+    return wrapper
+  }
+
+  beforeEach(() => {
+    wrapper = Wrapper()
+  })
+
+  describe('mount', () => {
+    it('starts with all categories button active', () => {
+      const allCategoriesButton = wrapper.find('.categories-filter .sidebar .base-button')
+      expect(allCategoriesButton.attributes().class).toContain('--filled')
+    })
+
+    it('sets category button attribute `filled` when corresponding category is filtered', async () => {
+      getters['posts/filteredCategoryIds'] = jest.fn(() => ['cat9'])
+      wrapper = await Wrapper()
+      democracyAndPoliticsButton = wrapper.findAll('.categories-filter .item .base-button').at(2)
```

Comment:
hey, thanks for the feedback! I agree with this, and have requested this change on other `PR`s as well, but I spent at least an hour, if not longer actually trying to get this to work, and it doesn't in this case because the button does not have a text, it has a separate `<label>`, and I couldn't workout how to filter the button based on it's sibling element.

What we could do is add a `data-test="categories-button-${category.slug}` and target the button that way. we remove `data-test` in production, so I think this would be better, what do you think?